app/javascript/chat/ActiveChatChannel/ChatMessages.js

```diff
@@ -0,0 +1,73 @@
+import { h } from 'preact';
+import PropTypes from 'prop-types';
+import ActionMessage from '../actionMessage';
+import Message from '../message';
+import DirectChatInfoMessage from './IntroductionMessages/DireactIntoMessage';
+import OpenChatInfoMessagge from './IntroductionMessages/OpenIntroMessage';
+
+const ChatMessages = ({
+  activeChannelId,
+  messages,
+  showTimestamp,
+  activeChannel,
+  currentUserId,
+  triggerActiveContent,
+  triggerEditMessage,
+  triggerDeleteMessage,
+}) => {
+  if (!messages[activeChannelId]) {
+    return '';
+  }
+
+  if (messages[activeChannelId].length === 0 && activeChannel) {
+    switch (activeChannel.channel_type) {
+      case 'direct':
+        return <DirectChatInfoMessage activeChannel={activeChannel} />;
+      case 'open':
+        return <OpenChatInfoMessagge activeChannel={activeChannel} />;
+      default:
+        return null;
+    }
+  }
+
+  return messages[activeChannelId].map((message) =>
+    message.action ? (
+      <ActionMessage
+        user={message.username}
+        profileImageUrl={message.profile_image_url}
+        message={message.message}
+        timestamp={showTimestamp ? message.timestamp : null}
+        color={message.color}
+        onContentTrigger={triggerActiveContent}
+      />
+    ) : (
+      <Message
+        currentUserId={currentUserId}
+        id={message.id}
+        user={message.username}
+        userID={message.user_id}
+        profileImageUrl={message.profile_image_url}
+        message={message.message}
+        timestamp={showTimestamp ? message.timestamp : null}
+        editedAt={message.edited_at}
+        color={message.color}
+        onContentTrigger={triggerActiveContent}
+        onDeleteMessageTrigger={triggerDeleteMessage}
+        onEditMessageTrigger={triggerEditMessage}
+      />
+    ),
+  );
+};
+
+Message.propTypes = {
+  activeChannelId: PropTypes.number,
+  messages: PropTypes.arrayOf(PropTypes.object),
```

Comment:
refactor (non-blocking): Consider using `PropTypes.shape` to describe the shape of a `message` prop. If the message proptype exist elsewhere, consider importing a shred prop type. For an example of this, see https://github.com/forem/forem/blob/master/app/javascript/common-prop-types/article-prop-types.js.

For more information on `PropTypes.shape`, see the [PropTypes](https://reactjs.org/docs/typechecking-with-proptypes.html) documentation.